난이도로 단어를 나누어 모델을 적용하는 구현

ene5135 opened the pull request:
난이도(엔트로피)로 단어를 나누어 높은 난이도의 단어에 svm을 적용하고 낮은 난이도의 단어에 필요에 따라 mfs를 적용하는 프로그램을 구현하였습니다.

모든 구현은 entropy_multi_model 디렉토리 안에서 진행하였습니다.

단어중의성 해소에 필요한 리소스를 만드는 스크립트(resource_builder.py)와 중의성 해소 및 평가를 진행하는 스크립트(multi_model_evaluator.py)를 분리하여 구현하였습니다.

각각 커맨드라인을 통해 각종 파라미터들을 받으며 자세한 구동 방법은 readme에 기록해두었습니다.

Commit:
난이도(엔트로피)별로 다른 모델을 적용하는 틀을 짭니다. #3

Commit:
엔트로피에 따른 모델 적용 #3

엔트로피에 따라 다른 모델을 적용하도록 구현하였습니다.
resource_builder.py 를 실행하면 모델 학습 및 평가에 의미 예측에 필요한
자료들을 생성합니다.

multi_model_evaluator.py 를 실행하면 학습된 모델을 바탕으로 입력받은
엔트로피 기준에 맞게 더 낮은 엔트로피의 단어이면 MFS로, 더 높은
엔트로피의 단어이면 SVM으로 의미를 예측합니다.

자세한 사항은 README에 남기겠습니다.

ene5135 committed:
엔트로피 분포 히스토그램 업로드 #3

Commit:
임베딩 방법별 학습된 svm 모델과 테스트 결과 업로드 #3

ulsan_glove : POS 태그가 있는 울산코퍼스로 glove 워드임베딩을 만들고
이를 피쳐로 학습한 svm모델(엔트로피가 0.1이상인 단어에 대해)

ulsan_w2v : POS 태그가 있는 울산코퍼스로 w2v 워드임베딩을 만들고
이를 피쳐로 학습한 svm모델(엔트로피가 0.1이상인 단어에 대해)

wiki_glove : POS 태그가 없는 위키덤프로 glove 워드임베딩을 만들고 이를
피쳐로 학습한 svm모델(엔트로피가 0.1이상인 단어에 대해)

Commit:
Merge branch 'feature/3' of https://github.com/kakaonlp-internship2018/wsd into feature/3

png 파일 충돌로 인한 머지

Commit:
코퍼스 전처리 단계에서 87, 88, 89, 99 등의 의미태그를 제거 #3

따라서 위 의미태그가 달려있던 단어는 동형이의어가 아닌것처럼 취급되고
점수계산에서 제외됩니다. 또한 svm 학습에서도 class에서 제외됩니다.

ene5135 committed:
Add files via upload

ene5135 committed:
Add files via upload

ene5135 committed:
Add files via upload

ene5135 committed:
Add files via upload

Commit:
resource_builder와 multi_model_evaluator 구현 #3

resource_builder : 모델학습에 필요한 각종 리소스를 구축하고 구축한
리소스를 바탕으로 모델을 학습합니다. 최종적으로 svm_dic.bin 파일에
학습된 svm 모델이 저장됩니다.

multi_model_evaluator : 준비된 svm 모델을 사용하여 테스트셋의 중의성
해소를 진행하고 평가합니다. 현재는 svm 단일로 사용하거나 낮은레벨에 MFS
모델을 적용할 수 있습니다.

statistic.py : 간단히 엔트로피 관련 통계정보를 뽑아볼 수 있는
스크립트입니다.

Commit:
Merge branch 'feature/3' of https://github.com/kakaonlp-internship2018/wsd into feature/3

png 파일로 인한 머지

ene5135 committed:
Update README.md

ene5135 committed:
Update README.md

Commit:
requirement 수정 #3

ene5135 committed:
Update README.md

krikit commented:
* glove 소스를 포함시키지 말고 git [submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules)을 활용하는 것이 좋을 것 같아요.
  - 혹은 [python package](https://pypi.org/project/glove/)를 활용하는 방법도 있어요.

* .png 파일들 중에 파일명에 한글이 들어간 경우 git이 오동작을 하는 것 같은데, 파일명은 영문으로 하는 것이 좋을 것 같아요.
  - 그리고, 소스 코드와 별로 상관이 없다면 다른 디렉토리에 모으는 것이 좋을 것 같습니다.

* 최종 학습된 glove/vectors.txt 파일은 너무 크지 않다면 압축해서 git 저장소에 포함하는 것도 좋을 것 같네요.
  - 이 결과를 위해 필요한 과정도 기술하면 더 좋을 것 같습니다.

* numpy, scipy, scikit-learn 패키지를 requirements.txt 파일에 추가해 줘야할 것 같아요.

* pylint의 defect 유형은 C, E, W, R이 있는데요, 각각 critical, error, warning, recommedation입니다. warning까지는 잡는 것이 좋을 것 같습니다.

Commit:
리뷰 조언들을 적용하였습니다. #3

- glove 소스를 제거하고 submodule로 포함시켰습니다.
- png 파일들의 이름을 영문으로 바꾸고 result 디렉토리로 묶었습니다.
- numpy, scipy, scikit-learn 패키지를 requirements.txt에 추가하였습니다.
- pylint warning level까지 통과하도록 코드를 수정하였습니다.
- vectors.txt 파일은 크기가 커서 따로 업로드 하지않고 생성방법을
readme에 서술하도록 하겠습니다.

Commit:
Merge branch 'feature/3' of https://github.com/kakaonlp-internship2018/wsd into feature/3

Commit:
png 파일 정리 #3

Commit:
README.md 수정 #3

ene5135 committed:
Update README.md

ene5135 commented:
리뷰 조언들을 적용하였습니다.
- glove 소스를 제거하고 submodule로 포함시켰습니다.
- png 파일들의 이름을 영문으로 바꾸고 result 디렉토리로 묶었습니다.
- numpy, scipy, scikit-learn 패키지를 requirements.txt에 추가하였습니다.
- pylint warning level까지 통과하도록 코드를 수정하였습니다.
- vectors.txt 파일은 크기가 커서 따로 업로드 하지않고 생성방법을 readme에 서술하였습니다.

서브모듈은 이런식으로 .gitmodules 파일만 생기면 되는건지 모르겠네요

리뷰 감사합니다!

Commit:
modify gitignore #3

Commit:
make_feature_vector 함수 내 list comprehension 수정 #3

make_feature_vector 함수 내 list comprehension 부분을 일반적인 for문으로
수정하였습니다.

Commit:
모호한 변수명 변경 #3
